src/common_main.c

```diff
@@ -102,8 +102,15 @@ void common_main(void)
 
     // securechip_setup must come after memory_setup, so the io/auth keys to be
     // used are already initialized.
-    if (!securechip_setup(&_securechip_interface_functions)) {
+    int securechip_result = securechip_setup(&_securechip_interface_functions);
```

Comment:
What about all the others? `securechip_update_keys`, `securechip_kdf` and so on could all propagate the error... although that might be a much bigger change.

Reply:
They don't lead to Aborts(), especially not at boot.

Of course it would be nice, but that is a long chain of propagation all the way to the host. I'd say out of scope for this PR.

Reply:
Yeah, that's what I thought too. Leave it for later.